distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Regex_Mode.enso

```diff
@@ -4,7 +4,7 @@
    to matching on the `Full` content of the input text.
 
 import project.Data.Numbers.Integer
-import project.Data.Text.Matching_Mode
+import project.Data.Text.Matching_Mode.Matching_Mode
```

Comment:
I cannot add a comment below but since we have merged `First` and `Last` to a common supertype, the line below must be updated accordingly:

```diff
- Bounded (start : Integer) (end : Integer) (mode : (Matching_Mode.First | Matching_Mode.Last | Regex_Mode) = Regex_Mode.All)
+ Bounded (start : Integer) (end : Integer) (mode : (Matching_Mode | Regex_Mode) = Regex_Mode.All)

Reply:
Once the `First` and `Last` are not standalone types, technically `Matching_Mode.First` cannot show up in the signature. It probably won't be verified yet, as our typesystem is practically nonexistent at this moment, but I think it would be good to at least try to keep ourselves up-to-date with the design (which currently disallows just a single variant of an ADT being used in type signatures).

So if you could please find usages of `Matching_Mode.First | Matching_Mode.Last` in type signatures and replace them with just `Matching_Mode`.

Also if anywhere we had just one of these variants - now this kind of assumption is inexpressible in the type signature (I think this was the reason why we had them be separate standalone types). If we have functions that assume just one variant, we probably need to:
1. Update the signature to be valid, i.e. take the "whole" `Matching_Mode`
2. Ideally, make sure that the code can handle the earlier unsupported matching mode gracefully - for example instead of inexhaustive pattern match, I'd suggest to throw an `Illegal_Argument.Error` explaining that the other mode is not supported.
3. Would be good to ensure the documentation makes it clear which modes are supported and which are not - since the type signature no longer provides this information.

I'm not sure if there even are instances where this is an issue, but please do check.

Reply:
It turns out there aren't any instances of this, but there was one false one, an unimplemented interface method, which probably should have had both First and Last in its signature. The single implementation of this interface method that was changed in this PR had both First and Last.